Sources/IO/Misc/OBJWriter/index.js

```diff
@@ -0,0 +1,179 @@
+import macro from 'vtk.js/Sources/macros';
+
+const { vtkErrorMacro } = macro;
+
+// ----------------------------------------------------------------------------
+// Global methods
+// ----------------------------------------------------------------------------
+
+// ----------------------------------------------------------------------------
+// vtkOBJWriter methods
+// ----------------------------------------------------------------------------
+
+const writeFaces = (faces, withNormals, withTCoords) => {
+  let outputData = '';
+  const fd = faces.getData();
+
+  let offset = 0;
+  while (offset < fd.length) {
+    const faceSize = fd[offset++];
+    outputData += 'f';
+    for (let i = 0; i < faceSize; i++) {
+      outputData += ` ${fd[offset + i] + 1}`;
+      if (withTCoords) {
+        outputData += `/${fd[offset + i] + 1}`;
+        if (withNormals) {
+          outputData += `//${fd[offset + i] + 1}`;
+        }
+      } else if (withNormals) {
+        outputData += `//${fd[offset + i] + 1}`;
+      }
+    }
+    offset += faceSize;
+    outputData += '\n';
+  }
+  return outputData;
+};
+
+const writeLines = (lines) => {
+  let outputData = '';
+  const ld = lines.getData();
+
+  let offset = 0;
+  while (offset < ld.length) {
+    const lineSize = ld[offset++];
+    outputData += 'l';
+    for (let i = 0; i < lineSize; i++) {
+      outputData += ` ${ld[offset + i] + 1}`;
+    }
+    offset += lineSize;
+    outputData += '\n';
+  }
+
+  return outputData;
+};
+
+const writePoints = (pts, normals, tcoords) => {
+  let outputData = '';
+  const nbPts = pts.getNumberOfPoints();
+
+  let p;
+
+  // Positions
+  for (let i = 0; i < nbPts; i++) {
+    p = pts.getPoint(i, p);
+    outputData += `v ${p[0]} ${p[1]} ${p[2]}\n`;
+  }
+
+  // Normals
+  if (normals) {
+    for (let i = 0; i < nbPts; i++) {
+      p = normals.getTuple(i, p);
+      outputData += `vn ${p[0]} ${p[1]} ${p[2]}\n`;
+    }
+  }
+
+  // Textures
+  if (tcoords) {
+    for (let i = 0; i < nbPts; i++) {
+      p = tcoords.getTuple(i, p);
+      outputData += `vt ${p[0]} ${p[1]}\n`;
+    }
+  }
+  return outputData;
+};
+
+const writeOBJ = (polyData) => {
```

Comment:
why not making it public ?

Reply:
i followed the same logic as in others Writers, it's exported in the default

Reply:
my bad :-)